### ISSUE TITLE
Prompt: Rewrite `quality-judge.md` with discrete rubrics, anchor examples, and bias mitigation

### ISSUE THREAD

**Issue by gregpriday** [canopyide/repo-expert#112]
## Description
The quality judge scores training examples on 4 dimensions. The current version has no rubrics, no anchor examples, and will produce inflated, uncalibrated scores. This prompt directly affects training data quality.

## Current State (22 lines)
- Lists 4 dimensions (relevance, completeness, realism, usefulness)
- Says "0-1 each" with no calibration
- No few-shot anchor examples
- No bias mitigation
- No reasoning-before-scoring

## What's Wrong
1. **Continuous 0-1 scores are uncalibrated** — LLMs default to 0.7-0.9 for everything
2. **No rubrics** — "relevance (0-1)" gives the model no behavioral anchors
3. **No anchor examples** — judge calibration drifts across examples
4. **Verbosity bias** — longer outputs get higher scores
5. **No reasoning structure** — model assigns scores without justification
6. **Batching temptation** — must be 1 evaluation per API call, never batched

### 1. Discrete rubric scores
Replace continuous 0-1 with 5-point discrete scale: {0.0, 0.25, 0.5, 0.75, 1.0}

### 2. Behavioral rubric for each dimension

**Relevance** (reference-based — needs ground truth):
```
1.00: All ground truth files appear in the output
0.75: Most ground truth files appear (>70%)
0.50: Some ground truth files appear (30-70%)
0.25: Few ground truth files appear (<30%)
0.00: No ground truth files appear in the output
```

**Completeness** (reference-based):
```
1.00: All major changed files AND tests are covered
0.75: All major source files covered, some tests missing
0.50: Primary files covered but secondary/dependent files missing
0.25: Only 1-2 correct files mentioned
0.00: No relevant files or tests mentioned
```

**Realism** (reference-free):
```
1.00: Guidance could have been written before seeing the PR diff
0.75: Mostly realistic, minor details suggest hindsight knowledge
0.50: Some recommendations clearly informed by knowing the final solution
0.25: Multiple recommendations reveal hindsight knowledge
0.00: The guidance is essentially a description of the final PR diff
```

**Usefulness** (reference-free):
```
1.00: Following this guidance would significantly accelerate implementation
0.75: Guidance provides meaningful direction with minor gaps
0.50: Guidance is partially helpful but missing key insights
0.25: Guidance is mostly generic/obvious, little repo-specific value
0.00: Guidance is misleading or would waste developer time
```

### 3. Add 2-3 anchor examples at different quality levels
- A "good" example (overall 0.75-1.0) with reasoning
- A "mediocre" example (overall 0.5) with reasoning explaining what's missing
- A "poor" example (overall 0.0-0.25) with reasoning explaining hallucination/leakage

### 4. Bias mitigation instructions
- "Do not penalize concise answers. Score based on information density, not length."
- "Do not reward verbose explanations that repeat the obvious."
- "Base your score strictly on the rubric definitions above."

### 5. Schema with reasoning first
```json
{
  "reasoning": "string (evaluate each dimension step by step)",
  "relevance_score": "number (enum: 0.0, 0.25, 0.5, 0.75, 1.0)",
  "completeness_score": "number",
  "realism_score": "number",
  "usefulness_score": "number",
  "overall_score": "number",
  "recommendation": "string (enum: 'good', 'acceptable', 'poor')"
}
```

### 6. Use the FULL model for judging
Always use gpt-5.4 (not mini) for quality scoring. The judge should be stronger than the teacher. Never judge a model with itself.

## Dependencies
Depends on #108 (system.md), #45 (JSON schemas)
